File: 2Star AdvLvl/ONEORTWO.py
def whoisWinner():
    x, y = map(int, input().split())

    dif = abs(x - y)
    if dif >= 2:
        if x > y:
            print("CHEF")
        else:
            print("CHEFINA")
    else:
        if x == y:
            if x % 2 == 1:
                print("CHEF")
            else:
                print("CHEFINA")
        else:
            mn = min(x, y)
            if mn % 2 == 1:
                print("CHEF")
            else:
                print("CHEFINA")

t = int(input())
for _ in range(t):
    whoisWinner()

File: 2Star AdvLvl/ONEORTWO.java
import java.util.Scanner;

public class Main {
    public static void whoisWinner() {
        Scanner scanner = new Scanner(System.in);
        int x = scanner.nextInt();
        int y = scanner.nextInt();

        long dif = Math.abs(x - y);
        if (dif >= 2) {
            if (x > y)
                System.out.println("CHEF");
            else
                System.out.println("CHEFINA");
        } else {
            if (x == y) {
                if (x % 2 == 1)
                    System.out.println("CHEF");
                else
                    System.out.println("CHEFINA");
            } else {
                long mn = Math.min(x, y);
                if (mn % 2 == 1)
                    System.out.println("CHEF");
                else
                    System.out.println("CHEFINA");
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while (t-- > 0) {
            whoisWinner();
        }
    }
}
